projects/mod_2_homework/src/main.c

```diff
@@ -0,0 +1,37 @@
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+typedef struct {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+static void timer_callback(SoftTimerId timer_id, void *ctx) {
+  Counters *ctr = ctx;
+  ctr->counter_a++;
+  LOG_DEBUG("Counter A: %i\n", ctr->counter_a);
+  if (ctr->counter_a % 2 == 0) {
+    ctr->counter_b++;
+    LOG_DEBUG("Counter B: %i\n", ctr->counter_b);
+  }
+  soft_timer_start_seconds(0.5, timer_callback, ctr, NULL);
```

Comment:
We usually ask that you factor out the timeout from here and the `soft_timer_start_seconds` call later into a #define (named descriptively and with the unit as a suffix of course). Also, it's more usual to use `soft_timer_start_millis` for timeouts that aren't a whole number of seconds due to the potential for floating point nonsense.

Reply:
Ah that makes sense; I'll change it 👍🏾